### PR TITLE
Fix aggressive ranged mobs not attacking from above a cliff

### PR DESCRIPTION
**Pull request by skyleo**
### Pull Request Prelude

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

Fix aggressive ranged mobs not attacking from above a cliff,
such as WoE Archer Guardians from above towers or Orc Archers in gef_fild14.

**Issues addressed:** none?

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

**Comment by HerculesWSAPI** [comment]
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/2550)

File: src/map/mob.c [hunk 1]
```diff
@@ -1173,13 +1173,15 @@ static int mob_ai_sub_hard_activesearch(struct block_list *bl, va_list ap)
...
-			if ((md->ud.walktimer == INVALID_TIMER && wpd.path_len > md->db->range2)
-				|| (md->ud.walktimer != INVALID_TIMER && wpd.path_len > md->db->range3))
-				return 0;
+				struct walkpath_data wpd;
+				bool is_standing = (md->ud.walktimer == INVALID_TIMER);
+				if (!path->search(&wpd, &md->bl, md->bl.m, md->bl.x, md->bl.y, bl->x, bl->y, 0, CELL_CHKNOPASS) // Count walk path cells
+				    || (is_standing && wpd.path_len > md->db->range2) //Standing monsters use range2, walking monsters use range3
+				    || (!is_standing && wpd.path_len > md->db->range3)) {
+					if (!check_distance_bl(&md->bl, bl, md->status.rhw.range)
+					    || !path->search_long(NULL, &md->bl, md->bl.m, md->bl.x, md->bl.y, bl->x, bl->y, CELL_CHKWALL))
```
Comment (4144) [inline src/map/mob.c]: this is official way?
if not, probably better add option and disable it by default

Comment (skyleo) [reply src/map/mob.c]: Well, the cells are marked as snipable, a player with a bow could attack from there as well. As far as I'm concerned Archer Guardians in WoE were always able to shoot from their watch towers.

Comment (bWolfie) [reply src/map/mob.c]: damn this bug existed since the dawn of woe. haha!!

Comment (Emistry) [reply src/map/mob.c]: I always thought it was map that blocking them from attacking from high grounds. lol

Comment (bWolfie) [reply src/map/mob.c]: me too @Emistry

**Pull request merged**